api/v1alpha1/gateway_types.go

```diff
@@ -180,6 +188,66 @@ type ListenerTLS struct {
 	Options map[string]string `json:"options"`
 }
 
+// TLSTerminationPolicy defines the schema of a TLS termination policy.
+type TLSTerminationPolicy struct {
+	// TerminationType defines how to terminate TLS connections.
+	//
+	// If unspecified, TLS termination type "Edge" will be used.
+	//
+	// Support: Core
+	//
+	// +optional
+	TerminationType TLSTerminationType `json:"terminationType,omitempty"`
+	// Routes is one or more routes to associate with the TLS termination policy.
+	//
+	// If unspecified, all routes associated with the Gateway listener will be used.
+	//
+	// Support: Core
+	//
+	// +optional
+	Routes []core.ObjectReference `json:"routes,omitempty"`
```

Comment:
@jpeach thanks for the review. This `Routes` field is meant to (optionally) provide the ability to associate the TLS termination policy to specific routes. This could be done in the `xRoute` resource, but then Devs would have that control instead of cluster admins. These routes could be all or a subset of routes that are bound to the gateway.

Reply:
I'm going to remove the route reference for now and make the termination policy a listener-scoped setting.